src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/ApproveReferredVacancyReviewCommandHandler.cs

```diff
@@ -3,46 +3,57 @@
-using System;
 using Esfa.Recruit.Vacancies.Client.Domain.Entities;
 using Esfa.Recruit.Vacancies.Client.Domain.Messaging;
 using Esfa.Recruit.Vacancies.Client.Domain.Events;
-using Esfa.Recruit.Vacancies.Client.Domain.Exceptions;
+using Microsoft.Extensions.Logging;
 
 namespace Esfa.Recruit.Vacancies.Client.Application.CommandHandlers
 {
     public class ApproveReferredVacancyReviewCommandHandler: IRequestHandler<ApproveReferredVacancyReviewCommand>
     {
+        private readonly ILogger<ApproveReferredVacancyReviewCommandHandler> _logger;
         private readonly IVacancyReviewRepository _vacancyReviewRepository;
         private readonly IVacancyRepository _vacancyRepository;
         private readonly IMessaging _messaging;
 
-        public ApproveReferredVacancyReviewCommandHandler(IVacancyReviewRepository vacancyReviewRepository, IVacancyRepository vacancyRespository, IMessaging messaging)
+        public ApproveReferredVacancyReviewCommandHandler(
+                            ILogger<ApproveReferredVacancyReviewCommandHandler> logger, 
+                            IVacancyReviewRepository vacancyReviewRepository, 
+                            IVacancyRepository vacancyRespository, 
+                            IMessaging messaging)
         {
+            _logger = logger;
             _vacancyReviewRepository = vacancyReviewRepository;
             _vacancyRepository = vacancyRespository;
             _messaging = messaging;
         }
 
         public async Task Handle(ApproveReferredVacancyReviewCommand message, CancellationToken cancellationToken)
         {
+            _logger.LogInformation("Approving referred review {reviewId} for vacancy {vacancyReference}.", message.ReviewId, message.Vacancy.VacancyReference);
+            
             var vacancy = await _vacancyRepository.GetVacancyAsync(message.Vacancy.Id);
 
             if (!vacancy.CanApprove)
             {
+                _logger.LogWarning($"Unable to approve review {{reviewId}} for vacancy {{vacancyReference}} due to vacancy having a status of {vacancy.Status}.", message.ReviewId, vacancy.VacancyReference);
                 return;
             }
-
-            await _vacancyRepository.UpdateAsync(message.Vacancy);
             
             var review = await _vacancyReviewRepository.GetAsync(message.ReviewId);
 
-            if (review.Status != ReviewStatus.UnderReview)
-                throw new InvalidStateException($"Review not in correct state to approve. State: {review.Status}");
+            if (!review.CanApprove)
+            {
+                _logger.LogWarning($"Unable to approve review {{reviewId}} for vacancy {{vacancyReference}} due to review having a status of {review.Status}.", message.ReviewId, vacancy.VacancyReference);
+                return;
+            }
 
             review.ManualOutcome = ManualQaOutcome.Approved;
             review.Status = ReviewStatus.Closed;
 
+            await _vacancyRepository.UpdateAsync(message.Vacancy);
```

Comment:
Do we need to do this I don't think it has changed? If we need to update I think we should be passing in vacancy not message.Vacancy